proto/poktroll/tokenomics/tx.proto

```diff
@@ -55,6 +55,10 @@ message MsgUpdateParam {
     string as_string = 3 [(gogoproto.jsontag) = "as_string"];
     int64 as_int64 = 6 [(gogoproto.jsontag) = "as_int64"];
     bytes as_bytes = 7 [(gogoproto.jsontag) = "as_bytes"];
+    // cannot add a map<string, int64> here
+    // do we want to create a custom message type
+    // or just store the chain-specific multipliers
```

Comment:
Add a type called:

```proto
message ServiceComputeUnitsToTokenMultiplers {
	map<string, int64> multipliers = 1;
}
```

Reply:
@Olshansk do we want the property named `ServiceComputeUnitsToTokenMultiplers` or `ServiceComputeUnitsToTokensMultiplers`? The previous property was called `compute_units_to_tokens_multiplier` with a plural `tokens` so I have thus far used that convention. Either works for me, I just want to be consistent.

Reply:
@Olshansk do we want the tokenomics param to use this type as well or should I just use this for update transactions? Currently, the `service_compute_units_to_tokens_multipliers` param is a raw `map<string, int64>`.

Reply:
This should be the end state:

```proto
message ComputeUnitsToTokensMultipliers {
	map<string, int64> service_compute_units_to_tokens_map = 1;
}

message Params {
	ComputeUnitsToTokensMultipliers compute_units_to_tokens_multipliers = 1;
}
```

I don't think `MsgUpdateParam` will be able to handle this though so CCing @bryanchriswhite for thoughts / input.


--- 

@rBurgett I don't fully understand the second question. Can you reiterate?

Reply:
@Olshansk yeah, `MsgUpdateParam` did not like that. You answered my questions with your comment above. I should be good to go. Thanks.

Reply:
@Olshansk It's not clear to me why you're suggesting to create a new type to hold the map. It seems to me that `Params` can have a map field for this purpose. However, if we do end up using a new type it MUST have the `option (gogoproto.equal = true);` option because `Params` has it and the generated code will not compile otherwise.

Reply:
@rBurgett If we can avoid the extra wrapper and have the `map` as a raw primtive in `Params`, that does make more sense. I don't recall how/why we went down the other path.

@bryanchriswhite How would the `Update` message be structure if we have a `map` in `Params`?

Reply:
The issue was the use of `oneof`  within the update message. A map cannot be used there. A raw map is fine in the tokenomics params. But, for the update transaction it needs to be something that can be used in `oneof`  in order to fit the current setup. So, we could serialize it as a string or wrap the map in a message, which you said was ideal. As of now, my testing with using a new message type in the update tx has not worked, with the final serialization failing with an error because it does not implement Message from the cosmos sdk. That is what I have been trying to figure out at the moment.

Reply:
I see the issue now with `MsgUpdateParam`. :thinking: Using a new type to encapsulate the map would be valid a workaround for this, although it might be a bit verbose.

```protobuf
// proto/poktroll/tokenomics/tx.proto
message MsgUpdateParam {
  ...
  oneof as_type {
    string as_string = 3 [(gogoproto.jsontag) = "as_string"];
    int64 as_int64 = 6 [(gogoproto.jsontag) = "as_int64"];
    bytes as_bytes = 7 [(gogoproto.jsontag) = "as_bytes"];
    StringInt64Map as_string_int64_map = 8 [(gogoproto.jsontag) = "as_string_int64_map"];
  }
}

message StringInt64Map {
  map<string, int64> entries = 1;
}
```

I don't see any reason why we wouldn't be able to continue to extend support for new types of params in this way.

Reply:
@bryanchriswhite for the moment, I have it implemented as @Olshansk suggested above, although I like the idea of generalizing it as you suggest. It is already verbose, so I don't see that as an issue. I am happy to implement it however you folks prefer.

Reply:
I would argue that if the reason for changing the data structure like this is to support `MsgUpdateParam`, then the additional types that result should be describing "generic" types that can be used in `as_type`. This would be as opposed to having use-case specific types, like @Olshansk's example.

I also think that these new types should be defined in the `shared` module's `params.proto` so that they can be reused between modules.

Reply:
`StringInt64Map` lgtm